test/test.js

```diff
@@ -169,6 +169,68 @@ tressa.async(function (done) {
   });
 })
 .then(function () {
+  return tressa.async(function (done) {
+    tressa.log('## preserve first child where first child is the same as incomming');
+    var div = document.body.appendChild(document.createElement('div'));
+    var render = hyperHTML.bind(div);
+    var observer = new MutationObserver(function (mutations) {
+      for (var i = 0, len = mutations.length; i < len; i++) {
+        trackMutations(mutations[i].addedNodes, 'added')
+        trackMutations(mutations[i].removedNodes, 'removed')
+      }
+    });
+
+    observer.observe(div, {
+      childList: true,
+      subtree: true,
+    });
+
+    var counters = [];
+
+    function trackMutations (nodes, countKey) {
+      for (var i = 0, len = nodes.length; i < len; i++) {
+        if (nodes[i] && nodes[i].getAttribute && nodes[i].getAttribute('data-test')) {
+          (counters[nodes[i].getAttribute('data-test')] || (counters[nodes[i].getAttribute('data-test')] = { added: 0, removed: 0 }))[countKey]++;
```

Comment:
Although fancy, could you maybe split this up a bit? Bit hard to follow. Also, nit, line length.